instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/SpringWebInstrumentationAutoConfigurationTest.java

```diff
@@ -11,26 +11,47 @@
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.autoconfigure.AutoConfigurations;
 import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.web.client.RestTemplate;
 
 class SpringWebInstrumentationAutoConfigurationTest {
 
   private final ApplicationContextRunner contextRunner =
       new ApplicationContextRunner()
           .withBean(OpenTelemetry.class, OpenTelemetry::noop)
+          .withBean(RestTemplate.class, RestTemplate::new)
           .withConfiguration(
               AutoConfigurations.of(SpringWebInstrumentationAutoConfiguration.class));
 
+  /**
+   * Tests the case that users create a {@link RestTemplate} bean themselves.
+   *
+   * <pre>{@code
+   * @Bean public RestTemplate restTemplate() {
+   *     return new RestTemplate();
+   * }
+   * }</pre>
+   */
   @Test
   void instrumentationEnabled() {
     contextRunner
         .withPropertyValues("otel.instrumentation.spring-web.enabled=true")
         .run(
-            context ->
-                assertThat(
-                        context.getBean(
-                            "otelRestTemplateBeanPostProcessor",
-                            RestTemplateBeanPostProcessor.class))
-                    .isNotNull());
+            context -> {
+              assertThat(
+                      context.getBean(
+                          "otelRestTemplateBeanPostProcessor", RestTemplateBeanPostProcessor.class))
+                  .isNotNull();
+
+              assertThat(
+                      context.getBean(RestTemplate.class).getInterceptors().stream()
```

Comment:
The test checks that the instrumentation implementation has added a RestTemplate interceptor. From a user perspective, a test verifying the spans would be more interesting.

Reply:
The interceptor itself is tested in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11054 - seems overkill to do it again for a slightly different style